alns/select/MABSelector.py

```diff
@@ -0,0 +1,201 @@
+import itertools
+from typing import List, Optional, Tuple
+
+import numpy as np
+from numpy.random import RandomState
+
+from alns.Outcome import Outcome
+from alns.State import ContextualState
+from alns.select.OperatorSelectionScheme import OperatorSelectionScheme
+
+MABWISER_AVAILABLE = True
+try:
+    from mabwiser.mab import MAB, LearningPolicy, NeighborhoodPolicy
+except ModuleNotFoundError:
+    MABWISER_AVAILABLE = False
+
+
+def ops2arm(destroy_idx: int, repair_idx: int) -> str:
+    """
+    Converts a tuple of destroy and repair operator indices to an arm
+    string that can be passed to self._mab.
+
+    Examples
+    --------
+    >>> ops2arm(0, 1)
+    "0_1"
+    >>> ops2arm(12, 3)
+    "12_3"
+    """
+    return f"{destroy_idx}_{repair_idx}"
+
+
+def arm2ops(arm: str) -> Tuple[int, int]:
+    """
+    Converts an arm string returned from self._mab to a tuple of destroy
+    and repair operator indices.
+
+    Examples
+    --------
+    >>> arm2ops("0_1")
+    (0, 1)
+    >>> arm2ops("12_3")
+    (12, 3)
+    """
+    [destroy, repair] = arm.split("_")
+    return int(destroy), int(repair)
```

Comment:
We start with the main content first if we we move these below the class. That's typically preferred. Can you move them down, below `MABSelector`?